src/model/services/controlplane/appcatalogs/appcatalogs.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpClient } from 'model/clients';
+import {
+  IAppCatalog,
+  IMetaData,
+  ISpec,
+  IStorage,
+} from 'model/services/controlplane/appcatalogs/types';
+import { getBaseConfiguration } from 'model/services/controlplane/base';
+
+/**
+ * Get the all the app catalogs in the installation.
+ * @param client - The HTTP client.
+ */
+export async function getAppCatalogs(
+  client: HttpClient
+): Promise<IAppCatalog[]> {
+  const { ApplicationGiantswarmIoV1alpha1Api } = await import(
+    'giantswarm-cp-client'
+  );
+  const baseConfig = await getBaseConfiguration(client);
+  const appsAPI = new ApplicationGiantswarmIoV1alpha1Api(baseConfig);
+
+  const catalogs = await appsAPI.listApplicationGiantswarmIoV1alpha1AppCatalog(
+    {}
+  );
+  const result = catalogs.items.map(convertAppCatalogCRToCatalog);
+
+  return result;
+}
+
+function convertAppCatalogCRToCatalog(
```

Comment:
This maps the App catalog CR type to the type that we use internally, so we can have control over what we store in memory.